src/test/java/org/javarosa/core/model/actions/OdkNewRepeatEventTest.java

```diff
@@ -0,0 +1,86 @@
+package org.javarosa.core.model.actions;
+
+import org.javarosa.core.test.Scenario;
+import org.javarosa.xform.parse.XFormParseException;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.isEmptyOrNullString;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.javarosa.test.utils.ResourcePathHelper.r;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Specification: https://opendatakit.github.io/xforms-spec/#the-odk-new-repeat-event.
+ */
+public class OdkNewRepeatEventTest {
+    @Test
```

Comment:
I think I'm coming back from the idea of statically declaring a form state in tests. I think now I'd write all the tests in a progressive way, imitating what a user would do while using Collect. I think it is closer to how JR gets used by clients and doesn't require so much hacking.

Having this into consideration, this test could be written like so:
```
@Test
public void setValueOnRepeatInsertInBody_setsValueInRepeat() {
    Scenario scenario = Scenario.init(r("event-odk-new-repeat.xml"));
    scenario.next(); // /data/my-toplevel-value
    scenario.next(); // Prompt new repeat
    scenario.createNewRepeat();

    assertThat(scenario.answerOf("/data/my-repeat[0]/defaults-to-now"), is(notNullValue()));
}
```

(`scenario.createNewRepeat();` just calls to `formEntryController.descendIntoNewRepeat();`)

Reply:
Agreed that's an improvement!